src/ip/QosConfig.cc

```diff
@@ -325,35 +326,30 @@ Ip::Qos::Config::parseConfigLine()
                        << "Netfilter mark preservation not available.");
 #endif // USE_LIBNETFILTERCONNTRACK
 #elif SO_MARK // SO_MARK && USE_LIBCAP
-                debugs(3, DBG_CRITICAL, "ERROR: Invalid parameter 'mark' in qos_flows option. "
-                       << "Linux Netfilter marking not available without LIBCAP support.");
-                self_destruct();
+                throw TextException(ToSBuf("ERROR: Invalid parameter 'mark' in qos_flows option. ", 
+                                           "Linux Netfilter marking not available on this platform."), Here());
 #else // SO_MARK && USE_LIBCAP
-                debugs(3, DBG_CRITICAL, "ERROR: Invalid parameter 'mark' in qos_flows option. "
-                       << "Linux Netfilter marking not available on this platform.");
-                self_destruct();
+                throw TextException(ToSBuf("ERROR: Invalid parameter 'mark' in qos_flows option. ", 
+                                           "Linux Netfilter marking not available on this platform."), Here());
 #endif // SO_MARK && USE_LIBCAP
             } else if (strncmp(token, "tos",3) == 0) {
                 preserveMissTos = true;
                 tos = true;
             } else {
-                preserveMissTos = true;
-                tos = true;
+                throw TextException(ToSBuf("Unsupported use case"), Here());
```

Comment:
If this code stays, please report the unsupported token value and, if possible, tell the admin what that token is (e.g. "qos_flows value type name"?) instead of using a generic "use case" description. Imagine an admin looking at a configuration line containing _multiple_ tokens, trying to understand which one is "unsupported"...

Reply:
Your right.
I will update this error message with meaningful description.
Thank you